modules/animation/include/modules/animation/datastructures/invalidationtrack.h

```diff
@@ -0,0 +1,103 @@
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <modules/animation/animationmoduledefine.h>
+
+#include <inviwo/core/io/serialization/deserializer.h>  // for ContainerWrapper<>::Item
+#include <inviwo/core/util/exception.h>                 // for Exception
+
+#include <modules/animation/datastructures/animationstate.h>  // for AnimationState, AnimationTi...
+#include <modules/animation/datastructures/animationtime.h>   // for Seconds
+#include <modules/animation/datastructures/basekeyframe.h>    // for BaseKeyframe
+#include <modules/animation/datastructures/basekeyframesequence.h>  // for BaseKeyframeSequence
+#include <modules/animation/datastructures/keyframe.h>              // for operator<
+#include <modules/animation/datastructures/basetrack.h>             // for BaseTrack<>::key_type
+
+#include <functional>  // for __base
+#include <memory>      // for unique_ptr
+#include <string>      // for basic_string
+#include <vector>      // for vector
+
+namespace inviwo {
+
+class ProcessorNetwork;
+
+namespace animation {
+
+class IVW_MODULE_ANIMATION_API InvalidationKeyframe : public BaseKeyframe {
+public:
+    using value_type = void;
+    InvalidationKeyframe() = default;
+    InvalidationKeyframe(Seconds time);
+    InvalidationKeyframe(const InvalidationKeyframe&);
+    InvalidationKeyframe& operator=(const InvalidationKeyframe&);
+    virtual ~InvalidationKeyframe();
+    virtual InvalidationKeyframe* clone() const override;
+
+private:
+};
+
+class IVW_MODULE_ANIMATION_API InvalidationKeyframeSequence
+    : public BaseKeyframeSequence<InvalidationKeyframe> {
+public:
+    InvalidationKeyframeSequence() = default;
+    InvalidationKeyframeSequence(std::vector<std::unique_ptr<InvalidationKeyframe>> keyframes);
+    InvalidationKeyframeSequence(const InvalidationKeyframeSequence& rhs) = default;
+    InvalidationKeyframeSequence& operator=(const InvalidationKeyframeSequence& that) = default;
+    InvalidationKeyframeSequence& operator=(InvalidationKeyframeSequence&& that) = default;
+    virtual ~InvalidationKeyframeSequence() = default;
+
+    virtual InvalidationKeyframeSequence* clone() const override;
+
+    virtual void serialize(Serializer& s) const override;
+    virtual void deserialize(Deserializer& d) override;
+
+    std::string processorId;
+};
+
+class IVW_MODULE_ANIMATION_API InvalidationTrack : public BaseTrack<InvalidationKeyframeSequence> {
```

Comment:
I think that the connection to what is invalidated could be more explicit as both properties and processors can be invalidated.
I am also not sure about the use case for this track. If a processor has not been invalidated there will not be any change in its output. Care to elaborate on this by documenting the class?
```suggestion
class IVW_MODULE_ANIMATION_API ProcessorInvalidationTrack : public BaseTrack<InvalidationKeyframeSequence> {
```

Reply:
If the name is changed I also think that corresponding changes should be made to the KeyFrame and the KeyFrameSequence